rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1268,15 +1280,24 @@ public BooleanWrapper updateDynamicParameters(String nodeSourceName, String infr
                                                                             oldDescriptor);
         persistNodeSourceWithNewDescriptor(newDescriptor);
 
-        // reconfiguration happens asynchronously
-        deployedNodeSource.reconfigure(newDescriptor.getInfrastructureParameters(),
-                                       newDescriptor.getPolicyParameters());
-
-        this.emitNodeSourceEvent(deployedNodeSource, RMEventType.NODESOURCE_UPDATED);
-
-        logger.info(NODE_SOURCE_STRING + nodeSourceName + " has been successfully updated with dynamic parameters");
-
-        return new BooleanWrapper(true);
+        try {
+            deployedNodeSource.reconfigure(newDescriptor.getInfrastructureParameters(),
+                                           newDescriptor.getPolicyParameters());
+            this.emitNodeSourceEvent(deployedNodeSource, RMEventType.NODESOURCE_UPDATED);
+            logger.info(NODE_SOURCE_STRING + nodeSourceName + " has been successfully updated with dynamic parameters");
+            return new BooleanWrapper(true);
+        } catch (Exception e) {
+            updateNodeSourceDescriptor(nodeSourceName,
+                                       deployedNodeSource,
+                                       oldInfrastructureParametersCopy,
+                                       oldPolicyParametersCopy);
+            persistNodeSourceWithNewDescriptor(oldDescriptor);
+            logger.warn(NODE_SOURCE_STRING + nodeSourceName +
+                        " failed to be updated with dynamic parameters. Infrastructure parameters: " +
+                        Arrays.toString(newDescriptor.getInfrastructureParameters()) + ". Policy parameters: " +
+                        Arrays.toString(newDescriptor.getPolicyParameters()) + ".", e);
+            throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)